tests/test_p4objects.cpp

```diff
@@ -517,6 +517,12 @@ class JsonBuilder {
     Json::Value cond(Json::objectValue);
     cond["name"] = name;
     cond["id"] = 0;
+    Json::Value source_info(Json::objectValue);
```

Comment:
this is not really what I had in mind, but we can add it as is and I will add a proper test after it is merged;,I will assign you for the review